extendr-api/src/lib.rs

```diff
@@ -6,59 +6,58 @@
 //!
 //! Anyone who knows the R library should be able to write R extensions.
 //! 
-//! See the [Robj](../robj/enum.Robj.html) struct for much of the content of this crate.
-//! [Robj](../robj/enum.Robj.html) provides a safe wrapper for the R object type.
+//! See [Robj] for much of the content of this crate.
+//! [Robj] provides a safe wrapper for the R object type.
 //! 
-//! This library is just being born, but goals are:
+//! Use attributes and macros to export to R.
+//! ```
+//! use extendr_api::*;
+//! // Export a function or impl to R.
+//! #[extendr]
+//! fn fred(a: i32) -> i32 {
+//!     a + 1
+//! }
 //!
-//! Implement common R functions such as c() and print()
+//! // define exports using extendr_module 
+//! extendr_module! {
+//!    mod mymodule;
+//!    fn fred;    
+//! }
+//! 
+//! ```
 //!
-//! Example:
+//! In R:
 //!
 //! ```ignore
-//! let v = c!(1, 2, 3);
-//! let l = list!(a=1, b=2);
-//! print!(v, l);
+//! result <- fred(1)
 //! ```
 //!
-//! Provide a wrapper for r objects.
+//! Common R functions such as c() and print() are available as macros:
 //!
-//! Example:
+//! ```
+//! use extendr_api::*;        // Put API in scope.
+//! extendr_engine::start_r(); // Start test environment.
+//! let v = c!(1., 2., 3.);
+//! let l = list!(a=1, b=2);
```

Comment:
Tidyverse R code style would place spaces around `=`: `a = 1, b = 2`

Reply:
Maybe we should follow rustfmt rather than tidyverse style as this is Rust code?

Reply:
Sounds reasonable. Is the convention no spaces under rustfmt?

Reply:
Seems so.

``` console
❯ echo 'fn f() {let l = list!(a=1, b=2);}' | rustfmt
fn f() {
    let l = list!(a = 1, b = 2);
}
```

Reply:
It seems we need spaces here anyway, sorry.

``` console
❯ echo 'fn f() {let l = list!(a=1, b=2);}' | rustfmt
fn f() {
    let l = list!(a = 1, b = 2);
}
```